main.go

```diff
@@ -9,26 +10,38 @@ import (
 	"github.com/datajet-io/peekaboo/alerting"
 	"github.com/datajet-io/peekaboo/config"
 	"github.com/datajet-io/peekaboo/services"
+	"github.com/cenk/backoff"
 )
 
 const configFilepath = "config.json" // path to config file
 const servicesfilePath = "services.json"
 
+
 //checks if there is Internet connection by pinging Google
 func hasInternet() bool {
+	operation := func() error {
+		// TODO: turn this into a config item
+		response, err := http.Get("https://www.google.com")
 
-	// Make request
-	response, err := http.Get("https://www.google.com")
+		if err != nil {
+			return err
+		}
 
-	if err != nil {
-		return false
-	}
+		defer response.Body.Close()
```

Comment:
we need to read a response body even if we are not going to use it. Like this: `io.Copy(ioutil.Discard, response.Body)`
http://stackoverflow.com/questions/17948827/reusing-http-connections-in-golang